dart/constraint/ContactConstraint.cpp

```diff
@@ -786,6 +812,57 @@ double ContactConstraint::computeSecondaryFrictionCoefficient(
   return dynamicAspect->getSecondaryFrictionCoeff();
 }
 
+//==============================================================================
+double ContactConstraint::computeSlipCompliance(
+    const dynamics::ShapeNode* shapeNode)
+{
+  assert(shapeNode);
+
+  auto dynamicAspect = shapeNode->getDynamicsAspect();
+
+  if (dynamicAspect == nullptr)
+  {
+    dtwarn << "[ContactConstraint] Attempt to extract slip compliance "
+           << "from a ShapeNode that doesn't have DynamicAspect. The default "
+           << "value (" << DART_DEFAULT_SLIP_COMPLIANCE << ") will be used "
+           << "instead.\n";
+    return DART_DEFAULT_SLIP_COMPLIANCE;
+  }
+
+  double slipCompliance = dynamicAspect->getSlipCompliance();
+  if (slipCompliance < 0)
```

Comment:
Correct me if I'm wrong. It seems a negative slip also makes sense, unlike friction. For example, a rotating tire could make slip either left or right.

Reply:
I understand slip compliance to be similar to a linear damping coefficient (actually the inverse, see [gazebo's WheelSlipPlugin documentation](https://github.com/osrf/gazebo/blob/gazebo11_11.2.0/plugins/WheelSlipPlugin.hh#L39-L77)), with a positive value corresponding to a dissipative effect. A negative value, on the other hand, would be non-dissipative and non-physical, similar to having a negative coefficient of friction.

Reply:
Yeah, I agree with @scpeters. In the rotating tire example, the tire can slip left or right, but that will be determined by the force applied on the tire. The slip velocity will always be in the same direction as the net external force. A negative slip parameter would mean a velocity opposite to the net external force, which seems non-physical.

Quoting from the ODE manual:
> FDS is an effect that causes the contacting surfaces to side past each other with a velocity that is proportional to the force that is being applied tangentially to that surface.

> Consider a contact point where the coefficient of friction mu is infinite. Normally, if a force f is applied to the two contacting surfaces, to try and get them to slide past each other, they will not move. However, if the FDS coefficient is set to a positive value k then the surfaces will slide past each other, building up to a steady velocity of kf relative to each other.